bee-message/src/payload/transaction/essence.rs

```diff
@@ -0,0 +1,381 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    constants::{INPUT_OUTPUT_COUNT_RANGE, IOTA_SUPPLY},
+    error::ValidationError,
+    input::Input,
+    output::Output,
+    payload::{Payload, PayloadPackError, PayloadUnpackError},
+};
+
+use bee_ord::is_sorted;
+use bee_packable::{
+    error::UnpackPrefixError, PackError, Packable, Packer, UnknownTagError, UnpackError, UnpackOptionError, Unpacker,
+    VecPrefix,
+};
+
+use alloc::vec::Vec;
+use core::{convert::Infallible, fmt};
+
+/// Length (in bytes) of Transaction Essence pledge IDs (node IDs relating to pledge mana).
+pub const PLEDGE_ID_LENGTH: usize = 32;
```

Comment:
this might be a silly suggestion but I'm realizing that we have many `NAME_LENGTH` constants. Maybe having a constant length should/could be a trait so instead of Having to do `TYPE_WITH_A_LONG_NAME_LENGTH` inside `TypeWithALongName` we could do `Self::LEN` or something.

Reply:
Agree with Christian. The name of those constants "TYPE_LENGTH" is begging for something like what you suggest.

Reply:
Alright, I'll make a trait. `FixedLength` sounds like it would work.